packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.Style.ts

```diff
@@ -0,0 +1,118 @@
+import { IDatePickerProps, IDatePickerStyles, IDatePickerStyleProps, DatePickerFormat } from './DatePicker.types';
+import { memoizeFunction } from '../../Utilities';
+import { concatStyleSets, IStyle } from '../../Styling';
+import { ResponsiveMode } from '../../utilities/decorators/withResponsiveMode';
+import { IComboBox, IComboBoxStyles } from 'src/index.bundle';
+
+export function getStyles(props: IDatePickerStyleProps): IDatePickerStyles {
+
+  const { displayDatePickerFormat } = props;
+  if (displayDatePickerFormat !== DatePickerFormat.dateOnly) {
+    return getDateAndTimePickerStyle(props);
+  }
+
+  // Inheriting from existing behavior, classname only override on root level style
+  const { responsiveMode, className, disabled, label } = props;
```

Comment:
move this into getDateOnly styles. i.e. make getStyles a gatekeeper, and the different themes each their own function. This makes the intent a bit more obvious.